### PR TITLE
Mangle compiled name for functions that cannot be called from langs o…

### PR DESCRIPTION
**Pull request by mausch**
…ther than F#

prevents https://github.com/hedgehogqa/fsharp-hedgehog/issues/156

### CODE REVIEW

**Review by moodmosaic — COMMENTED** [review]
Thank you for the pull request, @mausch. See https://github.com/hedgehogqa/fsharp-hedgehog/issues/156#issuecomment-703052451 first.

File: src/Hedgehog/Range.fs [hunk 1]
```diff
@@ -77,7 +77,7 @@ module Range =
 
     /// Construct a range which is unaffected by the size parameter using the
     /// full range of a data type.
-    [<CompiledName("ConstantBounded")>]
+    [<CompiledName("--ConstantBounded")>]
```
Comment (moodmosaic) [inline src/Hedgehog/Range.fs:80]: Is `--` some special trick that'll hide the member in other .NET languages? (I tried also `-` and it worked the same, I think.)

Comment (mausch) [reply src/Hedgehog/Range.fs:80]: One `-` would be the same yeah. It's just a way to set a name that's invalid for C# / VB.NET so you can't call it.

Comment (moodmosaic) [reply src/Hedgehog/Range.fs:80]: Should we use just one `-` then? (The simplest possible thing that gets the job done.)

Comment (mausch) [reply src/Hedgehog/Range.fs]: Sure why not. Changed in dbc14b4a4bf18a790e084532796e07e56faece07

Comment (moodmosaic) [reply src/Hedgehog/Range.fs]: :+1: :100:

**Comment by moodmosaic** [comment]
Alrighty, let's see if build passes.

**Comment by moodmosaic** [comment]
Build fails for some non-obvious reason. Let's investigate. :thinking:

**Comment by moodmosaic** [comment]
I'm not sure I can tell why build fails. Everything runs fine locally.

**Comment by moodmosaic** [comment]
Unsure why it fails on CI

```
Total tests: 182. Passed: 182. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 9.2574 Seconds
     1>Done Building Project "D:\a\fsharp-hedgehog\fsharp-hedgehog\Hedgehog.sln" (VSTest target(s)) -- FAILED.

Build FAILED.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:21.27
Error: Process completed with exit code 1.
```

**Review by moodmosaic — APPROVED** [review]
:+1: :beers:

**Pull request merged**